universal-application-tool-0.0.1/app/assets/javascripts/main.ts

```diff
@@ -151,86 +151,119 @@ function removeExistingEnumeratorField(event: Event) {
  */
 function removeLineClamp(event: Event) {
   const target = event.target as HTMLElement;
-  target.classList.add("line-clamp-none");
+  target.classList.add('line-clamp-none');
 }
 
 function attachLineClampListeners() {
   const applicationCardDescriptions = Array.from(document.querySelectorAll('.cf-application-card-description'));
-  applicationCardDescriptions.forEach(el => el.addEventListener("click", removeLineClamp));
+  applicationCardDescriptions.forEach(el => el.addEventListener('click', removeLineClamp));
 }
 
-/**
- * Filter the operators available for each scalar type based on the current scalar selected.
- */
-function filterOperators(event: Event) {
+function configurePredicateForm(event: Event) {
   // Get the type of scalar currently selected.
   const scalarDropdown = event.target as HTMLSelectElement;
   const selectedScalarType = scalarDropdown.options[scalarDropdown.options.selectedIndex].dataset.type;
+  const selectedScalarValue = scalarDropdown.options[scalarDropdown.options.selectedIndex].value;
 
+  filterOperators(scalarDropdown, selectedScalarType);
+  configurePredicateValueInput(scalarDropdown, selectedScalarType, selectedScalarValue);
+}
+
+/**
+ * Filter the operators available for each scalar type based on the current scalar selected.
+ */
+function filterOperators(scalarDropdown: HTMLSelectElement, selectedScalarType: string) {
   // Filter the operators available for the given selected scalar type.
   const operatorDropdown =
     scalarDropdown
-      .closest(".cf-predicate-options") // div containing all predicate builder dropdowns
-      .querySelector(".cf-operator-select") // div containing the operator dropdown
-      .querySelector("select") as HTMLSelectElement;
+      .closest('.cf-predicate-options') // div containing all predicate builder form fields
+      .querySelector('.cf-operator-select') // div containing the operator dropdown
+      .querySelector('select') as HTMLSelectElement;
 
   Array.from(operatorDropdown.options).forEach(option => {
     // Remove any existing hidden class from previous filtering.
-    option.classList.remove("hidden");
+    option.classList.remove('hidden');
     // If this operator is not for the currently selected type, hide it.
     if (!(selectedScalarType in option.dataset)) {
-      option.classList.add("hidden");
+      option.classList.add('hidden');
     }
   });
 }
 
+function configurePredicateValueInput(

```

Comment:
question (blocking): How does this work in the case where the value depends on the operator selected? For example, if I want to say `NUMBER is one of "12, 14, 16"`, the value input should be `text`, not `number`

Reply:
Oohh, great catch! Will implement.

Reply:
Discussed offline:

We actually don't have `is one of` and `is none of` currently configured as operators for scalar type LONG. That was kind of an accident, but it also seems like we can skip adding that feature right now while we try to prioritize the things we know we need and get them done by EOD Weds.

Reply:
Thanks Natalie! Added to #322 so others can work on it